### ISSUE TITLE
Question: is there a way to use wslg's xserver from another machine, like you would xming, vcxsrv ?

### ISSUE THREAD

**Issue by Ae3NerdGod** [microsoft/wslg#360]
ive been using xming to access gui applications on a linux server I have on my home network. is there a way to do the same via WSLg's xserver, or xwayland? itd be nice to ditch xming

**Comment by Biswa96** [comment]
wslg is using RDP which is different than X11 protocol. I think it's possible to streach that connection to another host.

**Comment by hideyukn88** [comment]
@Ae3NerdGod, thanks for suggestion, if you are looking for connecting to a machine running WSLg (say machine A) from another machine running Windows or any OS can run RDP client software (say machine B) ? So this allows the Linux GUI application (both X11 and Wayland) running on machine A will be projected (or remoted) to machine B's screen ? If so, this is possible to support with very minimal changes to our system-distro configuration (and port forwarding in WSL VM), from technical aspect. We will see how much people are interested in such usage. thanks!

**Comment by Ae3NerdGod** [comment]
> @Ae3NerdGod, thanks for suggestion, if you are looking for connecting to a machine running WSLg (say machine A) from another machine running Windows or any OS can run RDP client software (say machine B) ? So this allows the Linux GUI application (both X11 and Wayland) running on machine A will be projected (or remoted) to machine B's screen ? If so, this is possible to support with very minimal changes to our system-distro configuration (and port forwarding in WSL VM), from technical aspect. We will see how much people are interested in such usage, and we will need to evaluate its security model since WSLg is running on logged on user context on machine A, not separate service session. thanks!

im pretty sure you got it right
machine B has no xserver/wayland/display attached
normally id run something like DISPLAY=machineB.local:0.0 firefox, using an ssh client on machine A, the same as the old way of running GUI apps on WSL 1 and 2

so i guess id have to setup machineB to use machineA's display via an rdp connection rather than an xserver (if thats even possible)?

**Comment by hideyukn88** [comment]
@Ae3NerdGod, yes you are correct, in the proposed possible method, the protocol travels across two machine is RDP, not X11. Thus X server (and Wayland compositor) runs on machine A, which is also same physical machine runs Linux app. Then machine B only runs RDP client software (such as mstsc.exe) and connects to machine A using TCP.

**Comment by jimmyw** [comment]
I really think that this would be neat!

**Comment by khuongduybui** [comment]
@hideyukn88 piggy-backing on this issue, currently when I run `ssh -X` from inside an WSL2 distro, I get X11 forwarding to work, but running `ssh.exe -X` from Windows PowerShell does not work. Is there any trick we can pull here?

To be clear, I am using Visual Studio Code - Remote SSH extension, and I would like the terminal to have X11 forwarding, which is perfectly fine with XMing / Vscsrv / etc., but not WSLg.
